src/App.js

```diff
@@ -17,53 +16,42 @@ const App = () => {
     setPage(PAGE_HOME)
   }
 
+  const filterResults = data =>
+  data.filter(
+    item =>
+      !item.name.toLowerCase().includes('brewery in planning')
+  )
+
+  const fetchBreweryData = async () => {
+    const url = `https://api.openbrewerydb.org/breweries?by_${searchByParam}=${searchTerm}`
+    const data = await fetch(url)
+    const jsonData = await data.json()
+    const cleanData = filterResults(jsonData)
+    setBreweries(cleanData)
+  };
+
   useEffect(() => {
-    async function fetchBreweryData() {
-      const url = `https://api.openbrewerydb.org/breweries?by_${searchByParam}=${searchTerm}`
-      const data = await fetch(url)
-      const jsonData = await data.json()
-      const cleanData = filterResults(jsonData)
-      setBreweries(cleanData)
-    }
     fetchBreweryData()
-  })
+  }, [searchTerm, searchByParam])
 
   // searchTerm is the value from the input/search bar
-  const handleSearch = searchTerm => {
-    setSearchTerm(searchTerm)
+  const handleSearch = term => {
+    setSearchTerm(term)
     setPage(PAGE_RESULTS)
   }
 
   const searchBy = e => {
     setSearchByParam(e)
   }
 
-  // useEffect((previousProps, previousState) => {
-  //   const searchChanged = searchTerm !== previousState.searchTerm
-  //   if(searchChanged) {
-  //     this.fetchBreweryData(searchTerm, searchByParam)
-  //   }
-  // })
-
-
-//-------Original componentDidUpdate is below, unsuccessful attempt at hooks equivalent is above
-
-  // componentDidUpdate(previousProps, previousState) {
-  //   const { searchTerm, searchByParam } = this.state
-  //   const searchChanged = searchTerm !== previousState.searchTerm
-  //   if (searchChanged) {
-  //     this.fetchBreweryData(searchTerm, searchByParam)
-  //   }
-  // }
-
-  // Exclude breweries that contains "Brewery In Planning"
-  const filterResults = data =>
-    data.filter(
-      item =>
-        !item.name.toLowerCase().includes('brewery in planning')
-    )
+  useEffect((prevSearchTerm) => {
+    const searchChanged = searchTerm !== prevSearchTerm
+    if(searchChanged) {
+      fetchBreweryData(searchTerm, searchByParam)
+    }
+  }, [searchTerm, searchByParam])
 
```

Comment:
You don't have access to state in useEffect hook, thus prevSearchTerm will always be `undefined`. To get access to state you can use the useRef hook to keep a reference to the state. Place it near the top of the App function after you have defined the states: 

```js
import { Suspense, useEffect, useRef, useState } from 'react'
  [...]
  const usePrevious = value => {
    const ref = useRef()
    useEffect(() => {
      ref.current = value
    })
    return ref.current
  }

  const prevSearchTerm = usePrevious(searchTerm)
```
Now you have access to the previous search term, by including it as a dependency:

```js
  useEffect(() => {
    const searchChanged = searchTerm !== prevSearchTerm
    if (searchChanged) {
      ;(async () => {
        const url = `https://api.openbrewerydb.org/breweries?by_${searchByParam}=${searchTerm}`
        const data = await fetch(url)
        const jsonData = await data.json()
        const cleanData = filterResults(jsonData)
        setBreweries(cleanData)
      })()
    }
  }, [prevSearchTerm, searchTerm, searchByParam])
```

Notice I placed the fetchBreweryData function inside the useEffect hook (and made it an iife). This will remove the warning in console. I think I noticed that you have two useEffect hooks? One will suffice here.
Please also move the useEffect hook to be the last function right before the comoponent renders. This change is only to match previous standards where react lifecycles were usually placed near the bottom right before the render call.